contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_transform_resize.py

```diff
@@ -0,0 +1,370 @@
+        help="Value of height of the target resolution",
+    )
+
+    parser.add_argument(
+        "-f",
+        "--framerate",
+        type=str,
+        default="",
+        help="Value of framerate. e.g. 60, 30",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def get_golden_sample(
+    width: str = "3840",
+    height: str = "2160",
+    framerate: str = "60",
+    codec: str = "h264",
+    container: str = "mp4",
+) -> str:
+    """
+    Idealy, we can consume a h264 mp4 file then encode by any other codecs and
+    mux it with specific muxer such as mp4mux into mp4 container.
+    Therefore, we only need to adjust the width, height and framerate for
+    getting golden sample.
+    If you need a golden sample which doesn't exist in our sample pool, please
+    contribute it and get it as your requirement.
+    """
+    golden_sample = "big_bug_bunny_{}x{}_{}fps_{}.{}".format(
+        width, height, framerate, codec, container
+    )
+
+    full_path = os.path.join(
+        VIDEO_CODEC_TESTING_DATA, SAMPLE_2_FOLDER, golden_sample
+    )
+    logging.debug("Golden Sample: '{}'".format(full_path))
+    if not os.path.exists(full_path):
+        raise SystemExit(
+            "Error: Golden sample '{}' doesn't exist".format(full_path)
+        )
+
+    return full_path
+
+
+def execute_command(cmd: str) -> str:
+    """
+    Executes the GStreamer command and extracts the specific data from the
+    output. The specific data is the value of last-message which is exposed by
+    fpsdisplaysink.
+    :param cmd:
+        The GStreamer command to execute.
+    :returns:
+        The extracted last_message.
+    """
+    try:
+        logging.info("Starting command: '{}'".format(cmd))
+        ret = subprocess.run(
+            shlex.split(cmd),
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            encoding="utf-8",
+            timeout=300,
+        )
+        logging.info(ret.stdout)
+        return ret.stdout
+    except Exception as e:
+        raise SystemExit(e)
+
+
+def project_factory(**kwargs):
+    """
+    Factory
+    """
+    platform = kwargs.get("platform")
+    if "genio" in platform:
+        return GenioProject(**kwargs)
+    else:
+        raise SystemExit(
+            "Cannot find the implementation for '{}'".format(platform)
+        )
+
+
+class BaseHandler(ABC):
+    """Abstract base class for project-specific command handlers."""
+
+    def __init__(self, **kwargs):
+        self._platform = kwargs.get("platform")
+        self._codec = kwargs.get("encoder_plugin")
+        self._width_from = kwargs.get("width_from")
+        self._height_from = kwargs.get("height_from")
+        self._width_to = kwargs.get("width_to")
+        self._height_to = kwargs.get("height_to")
+        self._framerate = kwargs.get("framerate")
+        # Default extension is mp4, overwrite it if you need
+        self._output_file_extension = "mp4"
```

Comment:
I think this can be simplified if you pass in the entire `argparse.Namespace` from `parser.parse_args()` to this constructor and access the properties directly by their names. The existence of the property names like `args.width_from` is guaranteed if a default value or `required=True` is specified in the parser

Reply:
Fix